datastream-bigquery/src/main/java/com/linkedin/datastream/bigquery/BigqueryTransportProviderAdmin.java

```diff
@@ -116,13 +117,14 @@ public TransportProvider assignTransportProvider(final DatastreamTask task) {
                 throw new DatastreamRuntimeException("Unable to assign invalid datastream", e);
             }
         }
-        return _datastreamTransportProvider.computeIfAbsent(datastream, d -> {
+
+        final BigqueryTransportProvider transportProvider = _datastreamTransportProvider.computeIfAbsent(datastream, d -> {
```

Comment:
In this version we chose to instantiate a TP for each datastream. I like this approach to some extent as it provides a level of isolation per stream basis. However, it will impact how we do tuning and capacity planning.

I recommend we document the architecture for admins. We can discuss where this documentation can go.